src/Service/IndexData.php

```diff
@@ -134,7 +145,16 @@ function (IndexDataContextProvider $provider) {
             };
         }
 
-        $next();
+        try {
+            $next();
+        } finally {
+            static::$current = null;
+        }
+    }
+
+    public static function current(): ?self
```

Comment:
@chrispenny would you mind casting your eye over this - in discussions with @HeyImPhil we decided to add a 'current' call a bit like the current locale from fluent. This lets you get the active index data deeper down without having to pass the suffix info down every layer. Thoughts?